### ISSUE TITLE
SELU

### ISSUE THREAD

**Issue by michael-GitHub-0** [ageron/handson-ml#88]
Can you describe a bit about how the SELU scales change in different settings, such as with dropout and ConvNets?

Thanks!

Michael (michael.aurelien)

**Comment by ageron** [comment]
Hi Michael,

Great question. This is actually pretty well explained in the [Self-Normalizing Neural Networks paper](https://arxiv.org/pdf/1706.02515.pdf) that introduces SELU, I highly recommend reading it.

In short, the goal is to ensure that the variance and mean of each consecutive layer in the network remains constant during training and across layers, to avoid the exploding/vanishing gradients issue. The resulting neural net is said to be "Self-Normalizing". The default scale for the SELU activation function is tuned "just right" to ensure this property in a regular feedforward neural net (assuming the inputs have unit variance and zero mean). However, if the inputs have a different variance or mean, then you would need to tweak the SELU hyperparameters (but in practice, it's much simpler to normalize the inputs so that they have zero mean and unit variance).

Also, if you use regular dropout during training, then the neural network will no longer be Self-Normalizing even if you use the SELU activation function.  I won't get into the maths, since it's well explained in the paper, but basically with dropout the variance and mean end up changed at each layer. To avoid this, the paper introduces a variant of dropout, called *alpha dropout* that _is_ compatible with SELU activation and preserves the Self-Normalizing property of the network.
I have not looked into this much further yet, but I suppose that some architectures will require very specific hyperparameter values for the SELU activation function (i.e., tuning the scale and alpha) in order for the network to be self-normalized (e.g., what to do with a convolutional layer or a pooling layer, or with a recurrent neural net?).

Hope this helps!
Aurélien

**Comment by nmvega** [comment]
Splendid explanation. Thank you. (•◡•)/

**Comment by michael-GitHub-0** [comment]
Brilliant! I will read the article and get back to you if I put together anything cool. Aurelien is my new hero!!
